core/src/main/java/org/apache/iceberg/MetricsUtil.java

```diff
@@ -56,4 +63,125 @@ public static MetricsModes.MetricsMode metricsMode(
     String columnName = inputSchema.findColumnName(fieldId);
     return metricsConfig.columnMode(columnName);
   }
+
+  // Utilities for Displaying Metrics
+
+  static final Types.NestedField COLUMN_SIZES_METRICS =
+      optional(
+          300,
+          "column_sizes_metrics",
+          Types.MapType.ofRequired(301, 302, Types.StringType.get(), Types.LongType.get()),
+          "Map of column name to total size on disk");
+  static final Types.NestedField VALUE_COUNT_METRICS =
+      optional(
+          303,
+          "value_counts_metrics",
+          Types.MapType.ofRequired(304, 305, Types.StringType.get(), Types.LongType.get()),
+          "Map of column name to total count, including null and NaN");
+  static final Types.NestedField NULL_VALUE_COUNTS_METRICS =
+      optional(
+          306,
+          "null_value_counts_metrics",
+          Types.MapType.ofRequired(307, 308, Types.StringType.get(), Types.LongType.get()),
+          "Map of column name to null value count");
+  static final Types.NestedField NAN_VALUE_COUNTS_METRICS =
+      optional(
+          309,
+          "nan_value_counts_metrics",
+          Types.MapType.ofRequired(310, 311, Types.StringType.get(), Types.LongType.get()),
+          "Map of column name to number of NaN values in the column");
+  static final Types.NestedField LOWER_BOUNDS_METRICS =
+      optional(
+          312,
+          "lower_bounds_metrics",
+          Types.MapType.ofRequired(313, 314, Types.StringType.get(), Types.StringType.get()),
+          "Map of column name to lower bound in string format");
+  static final Types.NestedField UPPER_BOUNDS_METRICS =
+      optional(
+          315,
+          "upper_bounds_metrics",
+          Types.MapType.ofRequired(316, 317, Types.StringType.get(), Types.StringType.get()),
+          "Map of column name to upper bound in string format");
+  public static final Schema METRICS_DISPLAY_SCHEMA =
```

Comment:
Moved it here so I could access it for the TestHadoopSourceTableBase to construct the expected record.  Not sure if there is a better location.

Reply:
Can we make it package-private in the table and have tests in the corresponding module?